### PR TITLE
Techdebt/ab2d 4124 document enrollment details

### PR DESCRIPTION
**Pull request by sb-wnyffenegger**
**JIRA Tickets:**

[AB2D-4124](https://jira.cms.gov/browse/AB2D-4124) - Document Enrollment Details in Code
 
### What Does This PR Do?

Code Changes
* Remove deprecated database properties that are not in use currently
    * Remove from database using liquibase
    * Remove constants from properties service and constants
    * Adjust tests
* CoverageMappingCallabe use ContractType enum instead of hardcoded contract naming patterns
* Old properties not in use are removed from props files
* Improve code coverage for the CoverageDriverImpl so SonarQube passes

Documentation
* Document enrollment related objects CoveragePageRequest, CoveragePageResult
* CoverageServiceRepository - document details of each method with expected parameter values
     * pageCoverage methods are complex and are discussed in depth
     * insertCoverage method is discussed in depth because it is vanilla JDBC for a reason
* CoverageServiceImpl highlight business requirements handled by coverage service and why those requirements need to be carefully met
* ContractProcessorImpl - break down the complex parent method into the main steps occurring and the periodic actions being done to attempt to handle logging and issues like request errors.
     * Document details on how eob requests are created, queued, and processed in the main loop in loadEobRequests
* CoverageDriverImpl - document how concurrency is handled when multiple worker nodes are trying to run jobs and how the concurrently running methods work together to implement the workflow.
      * Define terms appearing for coverage like "stale" and "discovered"
      * Highlight locking mechanism that prevents concurrent issue
      * Explain how isCoverageAvailable determines whether coverage is available
      * Highlight business requirements that class is responsible for maintaining
* CoverageMappingCallable - step by step cover querying BFD, paging through results, and filtering
* CoverageProcessorImpl - most complex class in the worker so every detail is covered here
       * Different phases of searching for enrollment, collecting results, and inserting results into DB
       * Concurrency protection using synchronization around transitions between phases
       * External classes and what the functions inside those classes may be doing as side effects
       * How different failure cases are handled
 
### What Should Reviewers Watch For?

* Documentation that seems unclear or inaccurate and needs to be re-written or augmented
* Typos, incomplete thoughts

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/resources/application.properties [hunk 1]
```diff
@@ -25,17 +25,21 @@ eob.job.patient.queue.max.size=10000
...
+## ----------------------
+
+# Amount of time to wait to write to a file before throwing an error
 file.try.lock.timeout=30
-cancellation.check.frequency=1000
+
+# How often to write out % complete to the database (for display in status call to API)
 report.progress.db.frequency=100
+
+# How often to write out % complete to stdout
```
Comment (nflearl) [inline worker/src/main/resources/application.properties:37]: Put in the comment what the units are.  Are these msec or something else?

As a future aside, having the name of the property include the units is much clearer.  e.g. report.progress.db.frequency.ms=

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java [hunk 2]
```diff
@@ -486,36 +572,66 @@ private ZonedDateTime getStartDateTime(Job job) {
...
+    void checkCoveragePeriodValidity(Job job, CoveragePeriod period) {
         if (period.getStatus() == JobStatus.FAILED &&
                 period.getModified().isAfter(job.getCreatedAt())) {
             throw new CoverageDriverException("attempts to pull coverage information failed too many times, " +
                     "cannot pull coverage");
         }
     }
 
+    /**
+     * Page through coverage using a provided paging request
```
Comment (sharshar-sb) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java:584]: When you say "page", it wasn't obvious whether you meant in the database or BFD.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java [hunk 3]
```diff
@@ -486,36 +572,66 @@ private ZonedDateTime getStartDateTime(Job job) {
...
+            throw new CoverageDriverException("coverage driver failing preconditions", exception);
+        }
     }
 
+    /**
+     * Verify that coverage data cached in the database matches expected business requirements.
+     *
+     * Steps
+     *      - List of all contracts that are active contracts
+     *      - Filtered contracts {@link CoveragePeriodsPresentCheck}
```
Comment (sharshar-sb) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/coverage/CoverageDriverImpl.java:602]: Filtered contracts?

**Review by sharshar-sb — CHANGES_REQUESTED** [review]
Most of my comments are related to clarifying vague words such as "search", "results" or "business requirements". Is it a coverage database search, bfd search, etc.

**Review by sb-dkniesteadt — APPROVED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**